cpp/src/io/orc/orc_gpu.h

```diff
@@ -54,14 +56,15 @@ struct CompressedStreamInfo {
   }
   const uint8_t* compressed_data;  // [in] base ptr to compressed stream data
   uint8_t* uncompressed_data;  // [in] base ptr to uncompressed stream data or NULL if not known yet
-  size_t compressed_data_size;      // [in] compressed data size for this stream
-  gpu_inflate_input_s* decctl;      // [in] base ptr to decompression structure to be filled
-  gpu_inflate_status_s* decstatus;  // [in] results of decompression
-  gpu_inflate_input_s*
-    copyctl;  // [in] base ptr to copy structure to be filled for uncompressed blocks
+  size_t compressed_data_size;              // [in] compressed data size for this stream
+  device_span<uint8_t const>* dec_in_ctl;   // [in] input buffer to decompress
+  device_span<uint8_t>* dec_out_ctl;        // [in] output buffer to decompress into
+  decompress_status* decstatus;             // [in] results of decompression
+  device_span<uint8_t const>* copy_in_ctl;  // [out] input buffer to copy
+  device_span<uint8_t>* copy_out_ctl;       // [out] output buffer to copy to
   uint32_t num_compressed_blocks;  // [in,out] number of entries in decctl(in), number of compressed
```

Comment:
This is still needed, since this is the size of the array the dec_in_ctl points to. Since it's just a pointer, we have to pass the size separately (changing it from a pointer to span would open another can of worms).
Probably not apparent, but this comment indicates that `num_compressed_blocks` changes its meaning at some point, as it is the number of entries in `dec_in_ctl` going in, but it's the number of compressed blocks coming out :D
Cleaning this up deserves a dedicated effort IMO.